File: SSE_related_Code/matlab_Code/GPS_time_series_processing_code/read_seasonal_file.m
function seasonal = read_seasonal_file(filename)
%function write_seasonal_file(filename, seasonal)
%
%  Read the seasonal data structure seasonal from the file filename
%
dummy = struct('binval', [], 'smooth', []);

seasonal = struct( 'bintime', [], 'binsize', [], 'east', dummy, 'north', dummy, 'height', dummy);

input = load(filename, '-ascii');

nbins = size(input,1);

seasonal.binsize = input(1,2) - input(1,1);
seasonal.bintime = (input(:,2) + input(:,1))/2;

seasonal.east.smooth    = input(:,3);
seasonal.north.smooth   = input(:,4);
seasonal.height.smooth  = input(:,5);

seasonal.east.binval    = input(:,3);
seasonal.north.binval   = input(:,4);
seasonal.height.binval  = input(:,5);

return

File: SSE_related_Code/python_Code/GPS_processing_code/GPS_time_series_processing_code/read_seasonal_file.py
import numpy as np
# Read the seasonal data structure seasonal from the file filename
def read_seasonal_file(filename):  
    dummy = {'binval': [], 'smooth': []}

    seasonal = {
        'bintime': [],
        'binsize': [],
        'east': dummy.copy(),
        'north': dummy.copy(),
        'height': dummy.copy()
    }

    input_data = np.loadtxt(filename)

    nbins = input_data.shape[0]

    seasonal['binsize'] = input_data[0, 1] - input_data[0, 0]
    seasonal['bintime'] = (input_data[:, 1] + input_data[:, 0]) / 2

    seasonal['east']['smooth'] = input_data[:, 2]
    seasonal['north']['smooth'] = input_data[:, 3]
    seasonal['height']['smooth'] = input_data[:, 4]

    seasonal['east']['binval'] = input_data[:, 2]
    seasonal['north']['binval'] = input_data[:, 3]
    seasonal['height']['binval'] = input_data[:, 4]

    return seasonal
